Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -76,8 +75,8 @@ public MatchExtendedViewModel GetMatchForTeamByMatchID(int matchID)
         {
             var activity = _context.Activity
                 .Where(a => a.ID == _context.Series
-                    .FirstOrDefault(s => s.ID == _context.Match
-                        .FirstOrDefault(m => m.ID == matchID)
+                    .Find(_context.Match
+                        .Find(matchID)
                     .SeriesID)
                 .ActivityID)
```

Comment:
Couldn't you make this much more efficient by using navigation properties?:
```suggestion
            var activity = _context.Match.Find(matchID).Series.Activity
```

Reply:
Yes and no'ish. I just merged in the new implementation using status 0 as a deleted status (The thing I mentioned offline). One of the implementations is that I wanted to include a small version of the Activity which houses its ID and name since this function is used primarily for a small listing. If I were to return the whole activity, it might be too large of an object especially if we're calling 50-100 matches on a round robin for 10-20 teams. It might not matter when it gets launched on 360api1 but on our mini server, we can feel the performance dip if I return the whole object.

Reply:
If you only want the `ActivityID`, you could do
```suggestion
            var activity = _context.Match.Find(matchID).Series.ActivityID;
```

Reply:
Right, but I want the name and activityID.

Reply:
Maybe I'm not understanding you correctly. What performance issue are you concerned about here? You mention something about "returning the whole object", so are you saying you're going to completely re-write this to return just a subset of the activity data instead of the whole `ActivityExtendedViewModel`? 

Either way, I think the navigation properties are a nicer syntax than these nested `Find` calls, and at worst they should produce equivalent SQL queries.

Reply:
Actually, the more I think about it, this is a perfect usecase for the implicit operator. I actually will just return the extended viewmodel. Mentally, I just thought that that would entail every match + series + team + ...